core/bin/external_node/src/config/mod.rs

```diff
@@ -639,6 +608,13 @@ impl OptionalENConfig {
 /// This part of the external node config is required for its operation.
 #[derive(Debug, Deserialize, Clone, PartialEq)]
 pub(crate) struct RequiredENConfig {
+    /// L1 chain ID (e.g., 9 for Ethereum mainnet). This ID will be checked against the `eth_client_url` RPC provider on initialization
+    /// to ensure that there's no mismatch between the expected and actual L1 network.
+    pub l1_chain_id: L1ChainId,
+    /// L2 chain ID (e.g., 270 for zkSync Era mainnet). This ID will be checked against the `main_node_url` RPC provider on initialization
+    /// to ensure that there's no mismatch between the expected and actual L2 network.
+    pub l2_chain_id: L2ChainId,
```

Comment:
An alternative that would still preserve offline property (i.e., node being able to start / function w/o a stable connection to L1 or L2) is storing chain IDs in the node storage (as a part of `GenesisConfig`?). @Deniallugo Is there a plan to store `GenesisConfig` in the node storage eventually, or will it still be a (required?) part of the config? By @popzxc's classification of config params, `GenesisConfig` (or at least chain IDs) are immutable and set during genesis, so L1 is an authoritative source for the config; but if we want to support offline mode, we'd want to cache it locally.

Reply:
Yes, caching genesis configuration locally in the node storage matches my vision.